docs/specification.md

```diff
@@ -0,0 +1,596 @@
+
+## Swaps
+
+Users may call the `x_to_y` or `y_to_x` entrypoints to swap their `x` or `y` tokens, respectively.
+
+A tick is said to be _initialized_ if it is currently being used as a bound of a position,
+or _uninitialized_ otherwise.
+
+In the example above, we have two positions.
+Position `p1` ranges from `p(0)` to `p(4)`, and `p2` from `p(2)` to `p(6)`.
+
+```
+                             p2
+                  _______________________
+                 |                       |
+... p(0)  p(1)  p(2)  p(3)  p(4)  p(5)  p(6)  ...
+     |_______________________|
+                 p1
+```
+
+The ticks `p(0)`, `p(2)`, `p(4)` and `p(6)` are initialized,
+while the remaining are uninitialized.
+
+When a swap is initiated, the contract will first attempt to convert as many tokens as possible
+with the liquidity available up to the nearest initialized tick.
+While swapping inbetween two initialized ticks, the contract acts a constant product formula.
+
+For example, if the current price is `p(1)`, then the contract will convert as many tokens
+as possible using position `p1`'s liquidity alone.
+
+Once the price crosses an initialized tick, then the right amount of liquidity is
+added to/removed from the calculations.
+
+For example, when the price moves from `p(1)` to `p(2)`, we start taking position `p2`'s
+liquidity into account.
+When it moves from `p(3)` to `p(4)`, we stop taking `p1`'s liquidity into account.
+
+## Fees
+
+At the beginning of every swap, the contract subtracts a _swap fee_ from the tokens sent in,
+_before_ all other calculations. Whatever remains, is deposited into the liquidity pool and
+exchanged for the other asset.
+
+For example, if the swap fee is 0.3% and the user sends in 10000 `y` tokens, then:
+* 30 `y` tokens will be collected as swap fees and set aside,
+* 9970 `y` tokens will be deposited into the liquidity pool and exchanged for `x` tokens
+  according to the _constant function_ calculations.
+
+Furthermore, a _governance fee_ is subtracted from the swap fee.
+The governance fee is either expressed as a `1 / N` fraction or 0,
+and can be claimed by the contract's governance.
+Reusing the example above, if the governance fee were `1/5`, then `30/5 = 6` tokens would
+be subtracted from the swap fee.
+
+When an LP updates their position, they are rewarded with part of the swap fees paid by users,
+in both `x` and `y` tokens.
+This reward is proportional to:
+* the position's size,
+* and the amount of swap fees collected while the position remained _active_ since the last
+  time the position was updated/created.
+
+Both the swap fee and the governance fee percentages are initialized when the contract is
+originated and immutable thereafter.
+
+## Price Oracle
+
+Conceptually, at every block level, the contract will calculate a cumulative sum
+of the _current tick index_ `ic` since the contract's inception.
+
+At any point in time `t`, the acumulator `tick_cumulative(t)` is equal to:
```

Comment:
Since I'm reading the text, just some edits along the way

```suggestion
At any point in time `t`, the accumulator `tick_cumulative(t)` is equal to:
```